### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**
- Fixes unstable tests

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/Filter/CategoryFilterIntegration.php [hunk 1]
```diff
@@ -213,6 +213,7 @@ protected function createDataset(): void
             'values'     => ['color' => [['data' => 'blue', 'locale' => null, 'scope' => null]]],
         ]);
         $this->createProduct('unclassified-product', []);
+        sleep(1);
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/Filter/CategoryFilterIntegration.php]: quite ugly, but the only solution I could find to have ES find the related documents right after we insert them

Comment (grena) [reply tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/Filter/CategoryFilterIntegration.php]: cc @ahocquard 

Is your recent fix related to this kind of use case? Can't we call the refresh on clients?

Comment (mmetayer) [reply tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/Filter/CategoryFilterIntegration.php]: Oh, you're probably right, I'll try that and update the PR if it works

Comment (mmetayer) [reply tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/Filter/CategoryFilterIntegration.php]: Fixed

File: tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/SearchOnAttributesAndCategoriesIntegration.php [hunk 2]
```diff
@@ -315,6 +314,7 @@ protected function createDataset(): void
                 'description' => [['data' => 'quantum mechanics', 'locale' => null, 'scope' => null]],
             ],
         ]);
+        $this->get('akeneo_elasticsearch.client.product_and_product_model')->refreshIndex();
```
Comment (anaelChardan) [inline tests/back/Pim/Enrichment/Integration/ProductQueryBuilder/SearchOnAttributesAndCategoriesIntegration.php]: +1

**Comment by anaelChardan** [comment]
we made a lot of fixes here: https://github.com/akeneo/pim-community-dev/pull/9260/commits/65cdd8bb05d7adfca78cc50ee57414d2ff4b2601 maybe you could have a look to see if they are the same ;)

**Comment by mmetayer** [comment]
already fixed by #9260 :slightly_smiling_face:

**Pull request closed without merging**